pkg/cmd/build/new.go

```diff
@@ -70,33 +70,36 @@ func NewCmdBuildNew(f *factory.Factory) *cobra.Command {
 }
 
 func newBuild(org string, pipeline string, f *factory.Factory, message string, commit string, branch string, web bool) error {
-	l := io.NewPendingCommand(func() tea.Msg {
-		if len(branch) == 0 {
-			p, _, err := f.RestAPIClient.Pipelines.Get(org, pipeline)
-			if err != nil {
-				return err
+	var err error
+	var build *buildkite.Build
+	spinErr := spinner.New().
+		Title(fmt.Sprintf("Starting new build for %s", pipeline)).
+		Action(func() {
+			if len(branch) == 0 {
```

Comment:
What happens here if the `branch` is set to `   `? Do we trim the white space?

Reply:
Good catch I don't think it was handling that before but I'll add it now